### PR TITLE
Allow str and list in aggfunc in DataFrameGroupby.agg

### PR DESCRIPTION
**Pull request by charlesdong1991**
right now, when I look at Groupby, it does not accept str or list, but in pandas, it's allowed. So before implementing named aggregation, i think this is a better thing to deal first.
e.g. in pandas we could have:

Screen Shot 2019-09-24 at 10 58 52 PM

now koalas can also accept this:

Screen Shot 2019-09-24 at 11 00 13 PM

### CODE REVIEW

**Comment by charlesdong1991** [comment]
I am not sure if this will pass or not, since group key order is changed as you could see from the picture i attach. Any suggestions on a fix is welcome.

**Comment by charlesdong1991** [comment]
emm, even with current setup, `animals.groupby('kind').agg({'height': ['min', 'max']})` will have different order in index. Is it mandatory to have the exactly same output with pandas, the result looks correct except the order of rows changes. Probably should reorder the rows then before outputting the result. @ueshin @HyukjinKwon any thoughts?

**Comment by ueshin** [comment]
@charlesdong1991 We don't guarantee the row order without a special reason, e.g., `Series.value_counts` has a `sort` argument.

**Comment by charlesdong1991** [comment]
thanks for your comment, i slightly changed the test a bit to fix this order issue that failed tests, and added some docstrings in agg. Feel free to take a look. @ueshin

File: databricks/koalas/groupby.py [hunk 1]
```diff
@@ -112,14 +112,39 @@ def aggregate(self, func_or_funcs, *args, **kwargs):
...
+            if not isinstance(func_or_funcs, dict) or \
+                    not all(isinstance(key, (str, tuple)) and
+                            (isinstance(value, str) or isinstance(value, list) and
+                                all(isinstance(v, str) for v in value))
+                            for key, value in func_or_funcs.items()):
+                raise ValueError("aggs must be a dict mapping from column name (string or tuple) "
+                                 "to aggregate functions (string or list of strings).")
+
+        else:
+            func_dict = OrderedDict()
```
Comment (harupy) [inline databricks/koalas/groupby.py]: You could shorten this part to:
```
group_keyname = [key.name for key in self._groupkeys]
agg_cols = [key for key in self._kdf.columns if key not in group_keyname]
func_or_funcs = OrderedDict([(col, func_or_funcs) for col in agg_cols])
```

File: databricks/koalas/groupby.py [hunk 2]
```diff
@@ -112,14 +112,39 @@ def aggregate(self, func_or_funcs, *args, **kwargs):
...
+            group_keyname = [key.name for key in self._groupkeys]
+            agg_cols = [key for key in self._kdf.columns if key not in group_keyname]
+
+            for col in agg_cols:
+                func_dict[col] = func_or_funcs
+
+            func_or_funcs = func_dict
```
Comment (harupy) [inline databricks/koalas/groupby.py]: You could shorten this part to:
```
group_keyname = [key.name for key in self._groupkeys]
agg_cols = [key for key in self._kdf.columns if key not in group_keyname]
func_or_funcs = OrderedDict([(col, func_or_funcs) for col in agg_cols])
```

Comment (charlesdong1991) [reply databricks/koalas/groupby.py]: yeah, indeed! thx!

**Comment by charlesdong1991** [comment]
any follow-up review will be appreciated a lot ^^ @ueshin @HyukjinKwon

File: databricks/koalas/groupby.py [hunk 3]
```diff
@@ -112,14 +112,35 @@ def aggregate(self, func_or_funcs, *args, **kwargs):
...
+                    not all(isinstance(key, (str, tuple)) and
+                            (isinstance(value, str) or isinstance(value, list) and
+                                all(isinstance(v, str) for v in value))
+                            for key, value in func_or_funcs.items()):
+                raise ValueError("aggs must be a dict mapping from column name (string or tuple) "
+                                 "to aggregate functions (string or list of strings).")
+
+        else:
+            group_keyname = [key.name for key in self._groupkeys]
+            agg_cols = [key for key in self._kdf.columns if key not in group_keyname]
```
Comment (ueshin) [inline databricks/koalas/groupby.py]: I guess `self._agg_columns` should work?

Comment (charlesdong1991) [reply databricks/koalas/groupby.py]: Nice one! thanks! @ueshin

**Comment by ueshin** [comment]
@charlesdong1991 oh, one more thing I'd like to ask.
Could you add tests for multi-index columns as well, just in case?

**Comment by charlesdong1991** [comment]
@ueshin sure! Added and it passes tests locally, let's see if it's okay on CI.

**Comment by ueshin** [comment]
@charlesdong1991 Thanks!
LGTM again, pending tests.

File: databricks/koalas/tests/test_groupby.py [hunk 4]
```diff
@@ -151,6 +151,38 @@ def test_aggregate(self):
...
+                            'C': [0.362, 0.227, 1.267, -0.562]})
+        kdf = koalas.from_pandas(pdf)
+
+        columns = pd.MultiIndex.from_tuples([('X', 'A'), ('X', 'B'), ('Y', 'C')])
+        pdf.columns = columns
+        kdf.columns = columns
+
+        for aggfunc in agg_funcs:
+            sorted_agg_kdf = kdf.groupby('kind').agg(aggfunc)
+            sorted_agg_pdf = pdf.groupby('kind').agg(aggfunc)
```
Comment (ueshin) [inline databricks/koalas/tests/test_groupby.py]: We might need `.sort_index()`?

Comment (charlesdong1991) [reply databricks/koalas/tests/test_groupby.py]: emm, seems it has the same index as pandas when it's numeric. but could add one to ensure

**Comment by ueshin** [comment]
The latest failure seems not related to this PR.
Let me retrigger the tests.

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**